app/src/main/java/com/fastlink/zekrapp/presentation/ZekirScreen.kt

```diff
@@ -0,0 +1,121 @@
+package com.fastlink.zekrapp.presentation
+
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.pager.HorizontalPager
+import androidx.compose.foundation.pager.rememberPagerState
+import androidx.compose.material.FabPosition
+import androidx.compose.material.Scaffold
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowForward
+import androidx.compose.material.rememberScaffoldState
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.SnackbarHost
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalClipboardManager
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.fastlink.zekrapp.LocalNavController
+import com.fastlink.zekrapp.presentation.utils.AppBar
+import com.fastlink.zekrapp.presentation.utils.CustomFloatingActionButton
+import com.fastlink.zekrapp.presentation.utils.CustomSnackBar
+import com.fastlink.zekrapp.presentation.utils.ZekirCard
+import com.fastlink.zekrapp.presentation.utils.bottomAppBar.BottomBar
+import com.fastlink.zekrapp.presentation.utils.bottomAppBar.getListOfBottomBarItemsInZekirScreen
+import com.fastlink.zekrapp.viewModel.ZekirCategoryViewModel
+import com.fastlink.zekrapp.viewModel.ZekirViewModel
+import com.fastlink.zekrapp.viewModel.ZekirViewModelFactory
+
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun ZekirScreen(categoryId: Int) {
+    val categoryViewModel: ZekirCategoryViewModel = viewModel()
+    val zekirViewModel: ZekirViewModel = viewModel(factory = ZekirViewModelFactory(categoryId))
+    val snackBarState = remember { SnackbarHostState() }
+    val pagerState = rememberPagerState(
+        pageCount = { zekirViewModel.zekirs.size },
+    )
+    val scaffoldState = rememberScaffoldState()
+    val navController = LocalNavController.current
```

Comment:
LocalComposition is not used for everything. Read the documentation to see when you should use it:

https://developer.android.com/develop/ui/compose/compositionlocal

Here you can use rememberNavController instead.

Reply:
but if you change to this , created some issues

Reply:
the bottom App Bar don't show

Reply:
Then see why is this happening.

Reply:
Change this everywhere not only here.

Reply:
in this code 
`val navController = rememberNavController()
    val navBackStackEntry by navController.currentBackStackEntryAsState()`
returning the backstackEntry as null

Reply:
But don't call rememberNavController every time, pass the nacController to the parameter of every screen and then initialize with rememberNavController in your activity.

Reply:
and navigation between pages didn't work ?